src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+
+        for (const key of action.keysToRemove) {
+          if (key in state) {
+            delete state[key];
+          }
+        }
+        break;
```

Comment:
I'd recommend removing an empty line between case and for.
We can omit if (key in state).